src/gui/qgsrelationeditorwidget.cpp

```diff
@@ -379,17 +380,18 @@ void QgsRelationEditorWidget::updateUi()
   // If not yet initialized, it is not (yet) visible, so we don't load it to be faster (lazy loading)
   // If it is already initialized, it has been set visible before and the currently shown feature is changing
   // and the widget needs updating
-
   if ( mVisible )
   {
-    QgsFeatureRequest myRequest = mRelation.getRelatedFeaturesRequest( mFeature );
+    if ( mFeature.id() == mDualViewInitFeatureId )
+      return;
+    mDualViewInitFeatureId = mFeature.id();
```

Comment:
just in case it is assigned to another layer, should the feature id be cleared somewhere else in the code?

Reply:
@3nids , I don't think that's ever possible after the editor widget is built -- in fact, the parent attribute form doesn't expect the layer to change either, it's only set once in the constructor.

Reply:
thanks